bundles/statistics/statsgrid2016/components/Legend.js

```diff
@@ -58,63 +50,105 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.Legend', function(sandbox, loca
             return;
         }
         this._renderState.inProgress = true;
-        // start rendering
+
         var me = this;
-        var container = this.__element;
+        var container = this._element;
         var accordion = this._accordion;
-        // cleanup previous UI
         // NOTE! detach classification before re-render to keep eventhandlers
         this.editClassification.getElement().detach();
         accordion.removeAllPanels();
-        container.empty();
-        if(el) {
+
+        if ( container.find('.legend-noactive') ) {
```

Comment:
Find will always return an object so this will always be true. If you want to check if something was actually hit by the selector you need to check length of the returned object. Then again calling remove without the if-statement since you are not storing the result makes the performance worse.